### ISSUE TITLE
docusaurus: inject 'slug: /' in api/index.md

### ISSUE THREAD

**Issue by felipecrs** [tgreyuk/typedoc-plugin-markdown#143]
Since `homePageId: "api/index"` is deprecated, the `docs/api/index.md` should contain a new header:

```
---
slug: /
---
```

```log
The docs plugin option homePageId=api/index is deprecated. To make a doc the "home", prefer frontmatter: "slug: /"
```

**Comment by tgreyuk** [comment]
ok thanks will have a look at this.

**Comment by tgreyuk** [comment]
@felipecrs fixed  in 0.2.0

**Comment by felipecrs** [comment]
I see that now, `api/index` has the `slug: "/api"`. It makes sense, but what if the user wanted to use the `api/index` as the homepage (`slug: "/"`)?

I worked around it by [implementing my own `README.md` to `index.md` generation](https://github.com/felipecrs/megatar/pull/63), there would be many fields that the user would like to customize if you give them this option, so I'm not sure if new options to customize the `readme` in this plugin is really needed.

In any case, if in the future this feature gets implemented (https://github.com/facebook/docusaurus/issues/3475), we will be in business anyway.

**Comment by tgreyuk** [comment]
I see. I suppose we could provide another option `indexSlug` or something that defaults to `/api`.  I will leave this open for now.

**Comment by felipecrs** [comment]
> I see. I suppose we could provide another option indexSlug or something that defaults to /api. I will leave this open for now.

Beyond that, I would need to disable the `api/index` inclusion in the `typedoc-sidebar.js`, so I can put it in another category. Again, not sure if we should go that further in the customizations or not, I would like to hear more opinions.

**Comment by tgreyuk** [comment]
Would quite like to bottom this one out. Not sure if this will be an issue after #202 / #203 ?

**Comment by felipecrs** [comment]
Good point. In fact, after https://github.com/tgreyuk/typedoc-plugin-markdown/issues/203, I would need a new option: `readmeSlug: '/'`, otherwise I wouldn't be able to make it my homepage.

**Comment by felipecrs** [comment]
Also, just to you know, I tried to make docusaurus use the `typedoc-sidebar.js` just in case, and it fails:

```console
$ docusaurus start
Starting the development server...
Info: Loaded plugin typedoc-plugin-markdown
Info: TypeDoc sidebar written to /home/felipesantos/repos/megatar/website/typedoc-sidebar.js
Generated 7 of 7 TypeDoc docs
Info: Documentation generated at /home/felipesantos/repos/megatar/docs/api
Error: Error loading {"type":"category","collapsed":true,"label":"0","items":"a"}. "items" must be an array.
    at assertIsCategory (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:51:15)
    at normalizeItem (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:91:13)
    at arrayMap (/home/felipesantos/repos/megatar/node_modules/lodash.flatmap/index.js:140:21)
    at map (/home/felipesantos/repos/megatar/node_modules/lodash.flatmap/index.js:1928:10)
    at Object.flatMap [as default] (/home/felipesantos/repos/megatar/node_modules/lodash.flatmap/index.js:1881:22)
    at normalizeSidebar (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:114:36)
    at /home/felipesantos/repos/megatar/node_modules/lodash/lodash.js:13469:38
    at /home/felipesantos/repos/megatar/node_modules/lodash/lodash.js:4967:15
    at baseForOwn (/home/felipesantos/repos/megatar/node_modules/lodash/lodash.js:3032:24)
    at Function.mapValues (/home/felipesantos/repos/megatar/node_modules/lodash/lodash.js:13468:7)
    at normalizeSidebars (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:117:21)
    at Object.loadSidebars (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/sidebars.js:132:12)
    at loadVersion (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/index.js:98:45)
    at Array.map (<anonymous>)
    at Object.loadContent (/home/felipesantos/repos/megatar/node_modules/@docusaurus/plugin-content-docs/lib/index.js:150:68)
    at /home/felipesantos/repos/megatar/node_modules/@docusaurus/core/lib/server/plugins/index.js:55:59
```

**Comment by felipecrs** [comment]
Hey @tgreyuk, I just saw something at Docusaurs and I wanted to let you know:

https://github.com/facebook/docusaurus/pull/4485

Also, news at sidebars is incoming:

https://github.com/facebook/docusaurus/issues/3464#issuecomment-805088786

**Comment by tgreyuk** [comment]
Thanks for heads-up @felipecrs

**Comment by tgreyuk** [comment]
@felipecrs Can we close this ticket?

**Comment by samrith-s** [comment]
Idk what I'm doing wrong, but I get the slug as something weird:
```
slug: "//Users/samrith.shankar/Work/projects/concurrent-tasks/website/docs/api"
```
Because of this Docusaurus won't start. Any leads?

**Comment by tgreyuk** [comment]
I think this was because of changes in typedoc 0.21. If you try again with latest version it should work.

**Comment by edew** [comment]
Hey @tgreyuk 

Perhaps this is a separate issue but it would be great if we could specify the slug we want to use in `index.md` ourselves though a configuration option.

Would you be open to that?

**Comment by tgreyuk** [comment]
@edew yeah could provide an `indexSlug` option or something. will have a look.

**Comment by tgreyuk** [comment]
`indexSlug` option available will simply replace the slug value in index.md.

As for the original ticket the feature does appear to be doable as per https://github.com/facebook/docusaurus/issues/3475 .
